### DISCUSSION TITLE
Supabase Auth ASP.NET Core (.NET 8) WebAPI examples?

### DISCUSSION THREAD

**Discussion by mgardos01** [supabase-community/supabase-csharp#153] [Q&A]
Hi, 

I'm building a webapp right now that uses SvelteKit in the frontend and an ASP.NET Core Web API for the backend. I'd like to make use of as little SvelteKit backend functionality (```+[page,layout].server.[js,ts]```) as possible and relegate that to the Web API. Are there any public example projects out there that implement minimal Supabase Auth *specifically* with Web APIs I can build off of or learn from?

**Comment by acupofjose** [comment]
I (think) these may be close enough to what you're looking for to extrapolate from:
https://github.com/acupofjose/supasharp-todo/tree/master/SupasharpTodo.Shared/Providers
https://github.com/acupofjose/supasharp-todo/blob/master/SupasharpTodo.Shared/Extensions/ServiceCollectionExtensions.cs

**Comment by wiverson** [comment]
FWIW for SvelteKit the easiest thing to do IMHO is to use the regular Supabase JS client and set SvelteKit to SPA mode. That will help out with/simplify some of the SvelteKit context/stores. Then make sure you set the JS client to set a cookie with the Supabase JWT on login and/or add the JWT as a header to your REST calls or whatever. Then you can use this Supabase C# lib to make calls back to Supabase as needed, including auth verification.

Also FWIW I found that for most stuff it was really trivial to just put queries into Supabase itself and have them autoexposed as REST APIs directly using PostgREST. This kind of eliminated the need for a Java/C# middle tier, as I could write PL/SQL and/or a little bit of JS/TS instead. Just depends on what you are trying to do and the complexity of the server code.

**Reply by acupofjose** [reply]
^ very true. I found sveltekit in the backend to be super easy to work with. I think it’d be much faster to develop using the JS client than trying to implement the same functionality in the C#!
